src/core/meshcop/leader.cpp

```diff
@@ -91,7 +91,14 @@ void Leader::HandlePetition(Coap::Header &aHeader, Message &aMessage, const Ip6:
     SuccessOrExit(Tlv::GetTlv(aMessage, Tlv::kCommissionerId, sizeof(commissionerId), commissionerId));
     VerifyOrExit(commissionerId.IsValid());
 
-    VerifyOrExit(!mTimer.IsRunning());
+    if (mTimer.IsRunning())
+    {
+        VerifyOrExit(!strncmp(commissionerId.GetCommissionerId(), mCommissionerId.GetCommissionerId(),
+                              sizeof(mCommissionerId)));
+
+        mTimer.Stop();
```

Comment:
It's not the case for the call in HandleTimer(). Though it looks no harm, would it be good to call Timer.Stop() within a Timer handler? I'm okay with either way. :)